src/main/java/org/springframework/retry/support/RetryTemplate.java

```diff
@@ -590,6 +592,13 @@ private <T, E extends Throwable> void doCloseInterceptors(RetryCallback<T, E> ca
 		}
 	}
 
+	private <T, E extends Throwable> void doOnSuccessInterceptors(RetryCallback<T, E> callback, RetryContext context,
+			T result) {
+		for (int i = this.listeners.length; i-- > 0;) {
```

Comment:
Is there a clear view why we visit them in the reverse order?
I know: this comes as copy/paste from the `doOnErrorInterceptors()`, but still - "why?"

Thanks

Reply:
I don't know the reason; but it's clearly stated in the README.